cdap-watchdog/src/main/java/co/cask/cdap/logging/plugins/RollingLocationLogAppender.java

```diff
@@ -146,16 +153,40 @@ private void rollover(final LocationIdentifier logLocationIdentifier, ILoggingEv
             locationRollingPolicy.setLocation(locationOutputStream.getLocation(), new Closeable() {
               @Override
               public void close() throws IOException {
-                locationOutputStream.close();
-                locationManager.getActiveLocations().remove(logLocationIdentifier);
+                locationManager.getActiveLocations().remove(identifier);
+                try {
+                  locationOutputStream.close();
+                } catch (IOException e) {
+                  // If there is an exception while closing the outputstream, remember it and throw an exception so
+                  // that it can be closed on another event append
+                  locationManager.getInvalidOutputStreams().put(identifier, locationOutputStream);
```

Comment:
does this have to be a map ? there will be only one entry right ? as the last failed event will be kept retrying. can you just remember the last failed output stream in a variable ?